src/process_utility.c

```diff
@@ -472,6 +474,162 @@ process_drop_trigger_start(ProcessUtilityArgs *args, DropStmt *stmt)
 	ts_cache_release(hcache);
 }
 
+static void
+process_drop_procedure_start(ProcessUtilityArgs *args, DropStmt *stmt)
+{
+	ListCell *cell;
+
+	ScanIterator iterator =
+		ts_scan_iterator_create(BGW_JOB, RowExclusiveLock, CurrentMemoryContext);
+	ts_scanner_foreach(&iterator)
+	{
+		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
+		bool schema_isnull, name_isnull;
+		Name proc_schema =
+			DatumGetName(slot_getattr(ti->slot, Anum_bgw_job_proc_schema, &schema_isnull));
+		Name proc_name = DatumGetName(slot_getattr(ti->slot, Anum_bgw_job_proc_name, &name_isnull));
+		if (!schema_isnull && !name_isnull)
+		{
+			foreach (cell, stmt->objects)
+			{
+				ObjectWithArgs *object = castNode(ObjectWithArgs, lfirst(cell));
+				RangeVar *rel = makeRangeVarFromNameList(object->objname);
+				if (namestrcmp(proc_schema, rel->schemaname) == 0 &&
+					namestrcmp(proc_name, rel->relname) == 0)
+				{
+					Assert(stmt->removeType == OBJECT_PROCEDURE ||
+						   stmt->removeType == OBJECT_FUNCTION);
+					bool job_id_isnull;
+					const int32 job_id =
+						DatumGetInt32(slot_getattr(ti->slot, Anum_bgw_job_id, &job_id_isnull));
+					Ensure(!job_id_isnull, "job id was null");
+					ereport(ERROR,
+							errcode(ERRCODE_DEPENDENT_OBJECTS_STILL_EXIST),
```

Comment:
I guess we should adhere to CASCADE here and delete the job if CASCADE is given.

Reply:
Done.